spec/javascripts/actions/story_spec.js

```diff
@@ -549,4 +554,94 @@ describe('Story Actions', () => {
       });
     });
   });
+
+  describe('expandOrCollapseStory', () => {
+    const fakeDispatch = jest.fn();
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    const expandedStory = { id: 5 };
+    const collapsedStory = { id: 5, collapsed: true };
+    const dependencies = { Story: { deserialize: () => {} } };
+
+    describe('when is expanded', () => {
+      it('should dispatch toggleStory', async () => {
+        const thunkFunction = Story.expandOrCollapseStory(expandedStory, undefined);
+        await thunkFunction(fakeDispatch, null, dependencies);
+        expect(fakeDispatch).toHaveBeenCalledWith({
+          type: actionTypes.TOGGLE_STORY,
+          id: expandedStory.id,
+          from: undefined
+        });
+      });
+
+      it('should not dispatch "updateStorySuccess"', async () => {
+        const thunkFunction = Story.expandOrCollapseStory(expandedStory, undefined);
+        await thunkFunction(fakeDispatch, null, dependencies);
+        expect(fakeDispatch).not.toHaveBeenCalledWith({
+          type: actionTypes.UPDATE_STORY_SUCCESS,
+          story: expandedStory,
+          from: undefined
+        });
+      });
+
+      it('should not dispatch loading', async () => {
+        const thunkFunction = Story.expandOrCollapseStory(expandedStory, undefined);
+        await thunkFunction(fakeDispatch, null, dependencies);
+        expect(fakeDispatch).not.toHaveBeenCalledWith({
+          type: actionTypes.SET_LOADING_STORY,
+          id: expandedStory.id,
+          from: undefined,
+        });
+      });
+    });
+
+    describe('when is collapsed', () => {
+      it('should dispatch loading', async () => {
+        const thunkFunction = Story.expandOrCollapseStory(collapsedStory, undefined);
+        await thunkFunction(fakeDispatch, null, dependencies);
+        expect(fakeDispatch).toHaveBeenCalledWith({
+          type: actionTypes.SET_LOADING_STORY,
+          id: collapsedStory.id,
+          from: undefined,
+        });
+      });
+
+      describe('when request is successful', () => {
+        beforeEach(() => {
+          httpService.get.mockImplementationOnce(() => Promise.resolve({ data: { story: { } } }));
```

Comment:
Tip: You can use [`.mockResolvedValue()`](https://jestjs.io/pt-BR/docs/mock-function-api#mockfnmockresolvedvaluevalue) here

```suggestion
          httpService.get.mockResolvedValue({ data: { story: { } } });
```